python/nav/portadmin/snmp/base.py

```diff
@@ -67,6 +68,24 @@ def _wrapper(*args, **kwargs):
     return _wrapper
 
 
+class NoReadWriteManagementProfileError(GeneralException):
+    """No read-write management profile set on switch"""
+
+    pass
+
+
+class NoReadOnlyManagementProfileError(GeneralException):
+    """No read-only management profile set on switch"""
+
+    pass
+
+
+class InvalidManagementProfileError(GeneralException):
+    """Some attribute of management profile is incorrectly set"""
+
+    pass
+
+
```

Comment:
The Portadmin code base already features an exception hierarchy for management errors. You should inherit from these, rather than directly from `GeneralException`. The PortAdmin UI will not handle these exceptions gracefully.

See these:
https://github.com/Uninett/nav/blob/1e95655dcb8f0a1b10c18b108b16419e18388cf2/python/nav/portadmin/handlers.py#L282-L301